### DISCUSSION TITLE
请教关于GraphScope集群多次加载图数据执行图计算的问题

### DISCUSSION THREAD

**Discussion by TerryWang1208** [alibaba/GraphScope#1620] [Q&A]
您好，
最近搭了一个k8s集群，做了两个关于图计算的测试：
  - 加载一次数据，顺序执行100次图计算（包括page rank，wcc，bfs），顺利结束。
  - 每次图计算都重新加载数据，图计算结束时关闭session，重启session并加载数据做下一次计算。进行到第三次时，加载数据报错，中止。
针对这个现象，请教两个问题：
1. 关于测试#2中的现象，猜测可能是多次加载图数据导致内存不够。所以在session close时是否需要手动做内存资源释放？
2. 使用graphscope做图计算时，生产环境下，是否适合在每次图计算时，都临时拉起一个k8s来加载数据做计算，完成后再释放k8s资源？请建议。

谢谢！

**Comment by sighingnow** [comment]
> 关于测试#2中的现象，猜测可能是多次加载图数据导致内存不够。所以在session close时是否需要手动做内存资源释放？

不是预期行为，有可以复现的steps么

> 使用graphscope做图计算时，生产环境下，是否适合在每次图计算时，都临时拉起一个k8s来加载数据做计算，完成后再释放k8s资源？请建议。

视情况而定。假设两次图计算之间都相隔很久，那算完释放资源，需要的时候从新拉可能是一个更经济的做法。

**Reply by sighingnow** [reply]
当然如果比如读数据载图花费特别久，然后不同次的计算又是在同一个图上，hold集群不释放可能会有好。

**Comment by TerryWang1208** [comment]
感谢回复！
关于测试2的复现step，数据为LDBC SNB sf300的person和person-knows-person，graphscope版本为0.13.0。执行逻辑为外层通过一个循环，控制每次操作都加载一遍图，做一次图计算，关闭session，然后开始下一次循环。问题稳定出现在循环的第三次，加载图到60%时，报错。报错内容为：
creationDate: string
Loading vertex labeled person and edge labeled knows:  40%|████      | 4/10 [00:03<00:06,  1.02s/it]
Loading vertex labeled person and edge labeled knows:  60%|██████    | 6/10 [00:03<00:02,  1.59it/s]
2022-05-26 10:40:01,410 [DEBUG][utils:171]: Sleeping for 0.0625 before retrying failed request ...
Traceback (most recent call last):
  File "/data/terry/development/reps/GraphScope/python/graphscope/execNew/ldbc-test-all-data-load.py", line 52, in <module>
    pg = g.project(vertices={"person": []}, edges={"knows": []})
AttributeError: 'list' object has no attribute 'project'
再次连接集群执行算法时，会卡在以下位置，无法连接集群：
graphscope version is 0.13.0
2022-05-26 10:41:48,764 [INFO][session:678]: Connecting graphscope session with address: 192.168.0.157:30465



以下zip包括一个py文件和k8s的values.yaml，供参考。

期待回复，谢谢！

[code.zip](https://github.com/alibaba/GraphScope/files/8775888/code.zip)
